### PR TITLE
Form and search

### PR DESCRIPTION
**Pull request by mzlatic99**
Hello @kdelivuk @lukacevica @PetricMatej , I added possibility to add new place, edit it and delete it. The problem I have is myPlaces screen needs to rebuild to show new or edited place. In search part, I didn't used check in and check out in filtering results. They work but are not used because I don't have info on all available time for them. Can you please review and give feedback :)

### CODE REVIEW

File: lib/http.dart [hunk 1]
```diff
@@ -79,6 +79,27 @@ class HTTP {
         .map<Accommodation>((json) => Accommodation.fromJson(json))
         .toList();
   }
+
+  Future<Response> addNewPlace(Map<String, dynamic> map) {
+    return client.post(
+      'myplaces',
+      data: map,
+    );
+  }
```
Comment (PetricMatej) [inline lib/http.dart:88]: replace with :


    Future<List<Accommodation>> getMyPlaces() async {
    Response response = await client.get(
      'myplaces',
    );
    return response.data
        .map<Accommodation>((json) => Accommodation.fromJson(json))
        .toList();
  }

File: lib/screens/add_new_place-screen/add_new_place_screen.dart [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    final isValid = _formKey.currentState!.validate();
+    if (isValid) {
+      _formKey.currentState!.save();
+      edit
+          ? http.editPlace(int.parse(accommodation['id']), accommodation)
+          : http.addNewPlace(accommodation);
+      Navigator.of(context).pop();
+    }
+  }
+
```
Comment (PetricMatej) [inline lib/screens/add_new_place-screen/add_new_place_screen.dart:53]: add await 

  void _saveForm(bool edit) async {
    final isValid = _formKey.currentState!.validate();
    if (isValid) {
      _formKey.currentState!.save();
      edit
          ? await http.editPlace(int.parse(accommodation['id']), accommodation)
          : await http.addNewPlace(accommodation);
      await router.navigateTo(context, Routes.myPlacesScreen, null);
    }
  }

File: lib/http.dart (hunk 1 above)
Comment (mzlatic99) [reply lib/http.dart:88]: It's above addNewPlace :)

File: lib/screens/add_new_place-screen/add_new_place_screen.dart (hunk 2 above)
Comment (mzlatic99) [reply lib/screens/add_new_place-screen/add_new_place_screen.dart:53]: Works great, ty

**Review by lukacevica — APPROVED** [review]

**Pull request merged**